sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        The dictionary has three kewwords namely moment, slope and deflection.
+        The value of each keyword is a list of tuple, where each tuple
+        contains loaction and value of a boundary condition in the format
+        (location, value).
+
+        Examples
+        ========
+        There is a beam of length 4 meters. The bending moment at 0 should be 4
+        and at 4 it should be 0. The slope of the beam should be 1 at 0. The
+        deflection should be 2 at 0.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.bc_moment = [(0, 4), (4, 0)]
+        >>> b.bc_deflection = [(0, 2)]
+        >>> b.bc_slope = [(0, 1)]
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+
+        Here the deflection of the beam should be ``2`` at ``0``.
+        Similarly, the slope of the beam should be ``1`` at ``0`` and
+        there are two boundary conditions for bending moment. The bending
+        moment of the beam should be ``4`` at ``0`` and ``0`` at ``4``
+        """
+        return self._boundary_conditions
+
+    @property
+    def bc_moment(self):
+        return self._boundary_conditions['moment']
+
+    @bc_moment.setter
+    def bc_moment(self, m_bcs):
+        self._boundary_conditions['moment'] = m_bcs
+
+    @property
+    def bc_slope(self):
+        return self._boundary_conditions['slope']
+
+    @bc_slope.setter
+    def bc_slope(self, s_bcs):
+        self._boundary_conditions['slope'] = s_bcs
+
+    @property
+    def bc_deflection(self):
+        return self._boundary_conditions['deflection']
+
+    @bc_deflection.setter
+    def bc_deflection(self, d_bcs):
+        self._boundary_conditions['deflection'] = d_bcs
+
+    def apply_load(self, value, start, order, end=None):
+        """
+        This method adds up the loads given to a particular beam object.
+
+        Parameters
+        ==========
+        value :
+            The magnitude of an applied load.
+        start :
+            The starting point of the applied load.
+        end :
+            An optional argument that can be used if the load have an end point
+            within the length of the beam.
+        order :
+            The order of the applied load.
+
+            For Moments, order = -2
+            For Pointloads, order = -1
+            For constant distributed load, order = 0
+            For Ramp load, order = 1
+            For Parabolic Ramp, order = 2
+            ... so on.
```

Comment:
> The only issue with that is that we need a name for all orders to infinity.

I agree with this. For this reason, I think that order is much simpler than `type`.